test/test_python_dispatch.py

```diff
@@ -33,10 +32,15 @@ class LoggingTensor(torch.Tensor):
 
     @staticmethod
     def __new__(cls, elem, *args, **kwargs):
-        # The wrapping tensor (LoggingTensor) is just a meta tensor, so it
-        # doesn't hold any memory (meta tensor is generally the preferred type
-        # of tensor you want to make a subclass from)...
-        r = torch.Tensor._make_subclass(cls, elem.to(WRAPPER_DEVICE), elem.requires_grad)
+        # The wrapping tensor (LoggingTensor) shouldn't hold any
+        # memory for the class in question, but it should still
+        # advertise the same device as before
+        r = torch.Tensor._make_wrapper_subclass(
+            cls, elem.size(),
+            # TODO: clone strides and storage aliasing
+            dtype=elem.dtype, layout=elem.layout,
+            device=elem.device, requires_grad=elem.requires_grad
+        )
```

Comment:
What happens if someone calls .storage() on a LoggingTensor now?

Reply:
You get a storage, but its data pointer is a null pointer and alas, you can probably induce a segfault this way. Good catch.

Reply:
Indeed it segfaults

```
>>> class A(torch.Tensor):
...   def __torch_dispatch__(): pass
... 
>>> torch.Tensor._make_wrapper_subclass(A, (2,)).storage()[0]
Segmentation fault (core dumped)
```

I checked if meta tensors have this problem and they just error out when returning their storage, so I'll just do that here as a patch up